src/core/contourFeature.js

```diff
@@ -194,9 +198,55 @@ geo.contourFeature = function (arg) {
     if (gridW * gridH > data.length) {
       gridH = Math.floor(data.length) / gridW;
     }
+    /* If we are not using the position values (we are using x0, y0, dx, dy),
+     * and wrapLongitude is turned on, and the position spans 180 degrees,
+     * duplicate one or two columns of points at opposite ends of the map. */
+    usePos = (x0 === null || x0 === undefined || y0 === null ||
+        y0 === undefined || !dx || !dy);
+    if (!usePos && result.wrapLongitude && (x0 < -180 || x0 > 180 ||
+        x0 + dx * (gridW - 1) < -180 || x0 + dx * (gridW - 1) > 180) &&
+        dx > -180 && dx < 180) {
+      calcX = [];
+      for (i = 0; i < gridW; i += 1) {
+        x = x0 + i * dx;
+        while (x < -180) { x += 360; }
+        while (x > 180) { x -= 360; }
+        if (i && Math.abs(x - calcX[calcX.length - 1]) > 180) {
+          if (x > calcX[calcX.length - 1]) {
+            calcX.push(x - 360);
+            calcX.push(calcX[calcX.length - 2] + 360);
+          } else {
+            calcX.push(x + 360);
+            calcX.push(calcX[calcX.length - 2] - 360);
+          }
+          skipColumn = i;
+        }
+        calcX.push(x);
+      }
+      gridW += 2;
+      if (Math.abs(Math.abs(gridWorig * dx) - 360) < 0.01) {
+        gridW += 1;
+        x = x0 + gridWorig * dx;
+        while (x < -180) { x += 360; }
+        while (x > 180) { x -= 360; }
+        calcX.push(x);
+      }
+    }
+    /* Calculate the value for point */
     numPts = gridW * gridH;
     for (i = 0; i < numPts; i += 1) {
-      val = parseFloat(valueFunc(data[i]));
+      if (skipColumn === undefined) {
+        val = parseFloat(valueFunc(data[i]));
+      } else {
+        j = parseInt(Math.floor(i / gridW));
```

Comment:
My bad.  Here I keep wishing javascript had explicit data types and want to ensure it is an integer.